### DISCUSSION TITLE
Better support for dontStrip?

### DISCUSSION THREAD

**Discussion by mightyiam** [ipetkov/crane#510] [Q&A] [answered]
It seems that passing `dontStrip` does not suffice on its own, because the release cargo profile has — for one — `debug = false`. Would it make sense for crane to respond to `dontStrip` by overriding whatever cargo release profile options it takes to result in a useful backtrace?

**Comment by dpc** [comment, marked as answer]
IMO too much magic. Right now crane doesn't mess with Rust things more than absolutely needed, so it's relatively easy to know what's going on when you get the hang of the model.

**Reply by ipetkov** [reply]
It's also worth noting that there's so many different ways to influence cargo/rustc's behavior there aren't many good/robust ways of handling something like this, between:
* cargo environment variables
* RUSTFLAGS
* extra command line args passed to rustc from cargo
* cargo profiles in Cargo.toml
* cargo profiles in `.cargo/config.toml` anywhere up the file system hierarchy

So making it work is a challenge, much less avoiding surprises for why certain behavior happens/breaks. As annoying as it would be to apply this on a project-per-project basis, I think it would result in the least surprise/headaches for everyone involved
